src/Server/MySQLHandler.cpp

```diff
@@ -371,6 +384,79 @@ void MySQLHandler::comQuery(ReadBuffer & payload)
     }
 }
 
+void MySQLHandler::comStmtPrepare(DB::ReadBuffer & payload)
+{
+    if (prepared_statements_map.size() > 10000) /// Shouldn't happen in reality as COM_STMT_CLOSE cleans up the elements
+    {
+        LOG_ERROR(log, "Too many prepared statements");
+        current_prepared_statement_id = 0;
+        prepared_statements_map.clear();
+        packet_endpoint->sendPacket(ERRPacket(), true);
+        return;
+    }
+
+    String query;
+    readStringUntilEOF(query, payload);
+
+    uint32_t statement_id = current_prepared_statement_id;
+    if (current_prepared_statement_id == std::numeric_limits<uint32_t>::max())
```

Comment:
It will wrap around automatically if it's the max possible value minus one, no need for l. 402-407.